<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Looking into BrainXcan database</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">BrainXcan software documentation</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/liangyy/brainxcan-docs">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Looking into BrainXcan database</h1>

</div>


<div id="about" class="section level1">
<h1><span class="header-section-number">1</span> About</h1>
<p>In this post, we will look into the BrainXcan database and learn about how to load the following data into R:</p>
<ul>
<li>Prediction models</li>
<li>Genotype covariance</li>
<li>IDP GWASs</li>
</ul>
<p>These data loading rely on python packages <code>pandas</code> and <code>scipy.sparse</code> which has been installed as part of brainxcan conda environment. To use them in R, we can use <code>reticulate</code>.</p>
<p>The naming convention of IDP is <code>IDP-[UKB_Field_ID]</code> and we call it IDP ID. The information of the IDPs can be found at <a href="https://docs.google.com/spreadsheets/d/1xzLvwtIshyX10orGnGVyDNuyDHU-zgoYhGYemf-6I10">here</a>.</p>
<pre class="r"><code># load reticulate since we want to use python package
library(reticulate)</code></pre>
<pre><code>## Warning: package &#39;reticulate&#39; was built under R version 3.6.2</code></pre>
<pre class="r"><code># use brainxcan conda env
use_condaenv(&#39;brainxcan&#39;)
# load python packages
pd = import(&#39;pandas&#39;)
sp = import(&#39;scipy.sparse&#39;)</code></pre>
</div>
<div id="prediction-models" class="section level1">
<h1><span class="header-section-number">2</span> Prediction models</h1>
<p>We share prediction models of original IDPs and residual IDPs trained from ridge and elastic net. The residual IDPs are obtained by regressing out the first principal component of IDP matrix (by subtype) from the original IDP.</p>
<p>For each set of models, strutural IDPs and diffusion IDPs are saved separately and there are two types of files:</p>
<ul>
<li><code>*.parquet</code></li>
<li><code>*.perf.tsv.gz</code></li>
</ul>
<p>The parquet file contains the prediction model weights where SNPs are in rows and IDPs are in columns The prediction performance (obtained from training pipeline) of the models is in the <code>tsv.gz</code> file. If setting the number of rows in <code>tsv.gz</code> file as n, then the number of columns in the parquet is n + 4 with the four additional columns indicating: rsID (<code>snpid</code>), non effect allele (<code>a0</code>), effect allele (<code>a1</code>), and chromosome (<code>chr</code>).</p>
<p>The naming rule is:</p>
<ul>
<li><code>path-to-brainxcan_data/idp_weights/[model_type]/[idp_type].[modality].*</code>
<ul>
<li><code>model_type</code>: <code>ridge</code> or <code>elastic_net</code></li>
<li><code>idp_type</code>: <code>residual</code> or <code>original</code></li>
<li><code>modality</code>: <code>dmri</code> or <code>t1</code></li>
</ul></li>
</ul>
<p>Let’s take a look at them.</p>
<pre class="r"><code># set up the file path
model_type = &#39;ridge&#39;
idp_type = &#39;residual&#39;
modality = &#39;dmri&#39;
parquet_file = paste0(&#39;idp_weights/&#39;, model_type, &#39;/&#39;, idp_type, &#39;.&#39;, modality, &#39;.parquet&#39;)
perf_file = paste0(&#39;idp_weights/&#39;, model_type, &#39;/&#39;, idp_type, &#39;.&#39;, modality, &#39;.perf.tsv.gz&#39;)</code></pre>
<p>model_type = <code>ridge</code></p>
<p>idp_type = <code>residual</code></p>
<p>modality = <code>dmri</code></p>
<p>Reading parquet file: <code>path-to-brainxcan_data/idp_weights/ridge/residual.dmri.parquet</code></p>
<p>Reading performance file: <code>path-to-brainxcan_data/idp_weights/ridge/residual.dmri.perf.tsv.gz</code></p>
<pre class="r"><code># load the parquet file
df_weights = pd$read_parquet(paste0(datadir, &#39;/&#39;, parquet_file))
# load the performance file
df_perf = read.table(paste0(datadir, &#39;/&#39;, perf_file), stringsAsFactors = F, header = T) 

# look at the first 6 columns and rows
head(df_weights[, 1: 6]) %&gt;% pander</code></pre>
<table style="width:76%;">
<colgroup>
<col width="18%" />
<col width="6%" />
<col width="6%" />
<col width="8%" />
<col width="18%" />
<col width="18%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">snpid</th>
<th align="center">a0</th>
<th align="center">a1</th>
<th align="center">chr</th>
<th align="center">IDP-25650</th>
<th align="center">IDP-25651</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">rs3131972</td>
<td align="center">A</td>
<td align="center">G</td>
<td align="center">1</td>
<td align="center">-3.915e-05</td>
<td align="center">6.14e-05</td>
</tr>
<tr class="even">
<td align="center">rs3131969</td>
<td align="center">A</td>
<td align="center">G</td>
<td align="center">1</td>
<td align="center">-7.533e-05</td>
<td align="center">4.303e-05</td>
</tr>
<tr class="odd">
<td align="center">rs3131967</td>
<td align="center">T</td>
<td align="center">C</td>
<td align="center">1</td>
<td align="center">-7.558e-05</td>
<td align="center">4.256e-05</td>
</tr>
<tr class="even">
<td align="center">rs1048488</td>
<td align="center">C</td>
<td align="center">T</td>
<td align="center">1</td>
<td align="center">-3.991e-05</td>
<td align="center">5.655e-05</td>
</tr>
<tr class="odd">
<td align="center">rs12562034</td>
<td align="center">A</td>
<td align="center">G</td>
<td align="center">1</td>
<td align="center">-7.253e-07</td>
<td align="center">-7.05e-05</td>
</tr>
<tr class="even">
<td align="center">rs12124819</td>
<td align="center">G</td>
<td align="center">A</td>
<td align="center">1</td>
<td align="center">-3.426e-06</td>
<td align="center">1.991e-05</td>
</tr>
</tbody>
</table>
<pre class="r"><code>head(df_perf) %&gt;% pander</code></pre>
<table style="width:60%;">
<colgroup>
<col width="13%" />
<col width="13%" />
<col width="15%" />
<col width="16%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">R2</th>
<th align="center">Pearson</th>
<th align="center">Spearman</th>
<th align="center">phenotype</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">0.01224</td>
<td align="center">0.1107</td>
<td align="center">0.1094</td>
<td align="center">IDP-25650</td>
</tr>
<tr class="even">
<td align="center">0.01266</td>
<td align="center">0.1125</td>
<td align="center">0.1102</td>
<td align="center">IDP-25651</td>
</tr>
<tr class="odd">
<td align="center">0.02177</td>
<td align="center">0.1476</td>
<td align="center">0.1409</td>
<td align="center">IDP-25652</td>
</tr>
<tr class="even">
<td align="center">0.01806</td>
<td align="center">0.1346</td>
<td align="center">0.1313</td>
<td align="center">IDP-25653</td>
</tr>
<tr class="odd">
<td align="center">0.01723</td>
<td align="center">0.1325</td>
<td align="center">0.1285</td>
<td align="center">IDP-25654</td>
</tr>
<tr class="even">
<td align="center">0.02</td>
<td align="center">0.143</td>
<td align="center">0.1363</td>
<td align="center">IDP-25655</td>
</tr>
</tbody>
</table>
<p>Number of rows in weights table = 1071343</p>
<p>Number of columns in weights table = 312</p>
<p>Number of rows in performance table = 308</p>
<p>The parquet file support reading by column. So, if we only want to load the weights of a specific IDP, we can do the following:</p>
<pre class="r"><code>idp_id = &#39;IDP-25652&#39;
df_weights_sub = pd$read_parquet(
  paste0(datadir, &#39;/&#39;, parquet_file), 
  columns = c(&#39;snpid&#39;, &#39;a0&#39;, &#39;a1&#39;, &#39;chr&#39;, idp_id)
) 
df_weights_sub %&gt;% head %&gt;% pander</code></pre>
<table style="width:58%;">
<colgroup>
<col width="18%" />
<col width="6%" />
<col width="6%" />
<col width="8%" />
<col width="18%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">snpid</th>
<th align="center">a0</th>
<th align="center">a1</th>
<th align="center">chr</th>
<th align="center">IDP-25652</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">rs3131972</td>
<td align="center">A</td>
<td align="center">G</td>
<td align="center">1</td>
<td align="center">2.21e-06</td>
</tr>
<tr class="even">
<td align="center">rs3131969</td>
<td align="center">A</td>
<td align="center">G</td>
<td align="center">1</td>
<td align="center">1.576e-05</td>
</tr>
<tr class="odd">
<td align="center">rs3131967</td>
<td align="center">T</td>
<td align="center">C</td>
<td align="center">1</td>
<td align="center">1.627e-05</td>
</tr>
<tr class="even">
<td align="center">rs1048488</td>
<td align="center">C</td>
<td align="center">T</td>
<td align="center">1</td>
<td align="center">4.518e-06</td>
</tr>
<tr class="odd">
<td align="center">rs12562034</td>
<td align="center">A</td>
<td align="center">G</td>
<td align="center">1</td>
<td align="center">-0.0001214</td>
</tr>
<tr class="even">
<td align="center">rs12124819</td>
<td align="center">G</td>
<td align="center">A</td>
<td align="center">1</td>
<td align="center">-4.941e-05</td>
</tr>
</tbody>
</table>
</div>
<div id="genotype-covariance" class="section level1">
<h1><span class="header-section-number">3</span> Genotype covariance</h1>
<p>Genotype covariance was computed using the UKB IDP cohort (the same as prediction model training). The genotype covariance is saved by chromosome. It is a banded matrix with the band width = 200 and only the upper triangular values are saved since it is symmetric. These genotype covariance files are at <code>brainxcan_data/geno_covar/</code> and for each chromosome, there are two files:</p>
<ul>
<li><code>chr*.banded.npz</code>: The banded genotype covariance in scipy sparse matrix format.</li>
<li><code>chr*.banded.snp_meta.parquet</code>: The meta data annotating the rows/columns of the genotype covariance file in parquet format. The number of rows of the meta data matches the genotype covariance file and <strong>the SNPs are in the same order</strong>.</li>
</ul>
<pre class="r"><code># load banded genotype covariance 
mat = sp$load_npz(paste0(datadir, &#39;/geno_covar/chr22.banded.npz&#39;))

# take the first 500 SNPs and visualize
mat_sub = as.matrix(mat[ 1 : 500, 1 : 500])
var_sub = diag(mat_sub)
cor_sub = sweep(mat_sub, 1, FUN = &#39;/&#39;, sqrt(var_sub))
cor_sub = sweep(cor_sub, 2, FUN = &#39;/&#39;, sqrt(var_sub))
par(mfrow = c(1, 2))
image(mat_sub, main = &#39;Covariance between SNPs&#39;)
image(cor_sub, main = &#39;Correlation between SNPs&#39;)</code></pre>
<p><img src="database_files/figure-html/unnamed-chunk-7-1.png" width="960" /></p>
<p>class(mat) = dgTMatrix</p>
<p>To annotate the genotype covariance with the actual SNP information, we should load the meta data.</p>
<pre class="r"><code>df_snp = pd$read_parquet(paste0(datadir, &#39;/geno_covar/chr22.banded.snp_meta.parquet&#39;))
df_snp %&gt;% head %&gt;% pander</code></pre>
<table style="width:76%;">
<colgroup>
<col width="18%" />
<col width="22%" />
<col width="27%" />
<col width="8%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">snpid</th>
<th align="center">effect_allele</th>
<th align="center">non_effect_allele</th>
<th align="center">chr</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">rs7287144</td>
<td align="center">G</td>
<td align="center">A</td>
<td align="center">22</td>
</tr>
<tr class="even">
<td align="center">rs5748662</td>
<td align="center">A</td>
<td align="center">G</td>
<td align="center">22</td>
</tr>
<tr class="odd">
<td align="center">rs5747620</td>
<td align="center">C</td>
<td align="center">T</td>
<td align="center">22</td>
</tr>
<tr class="even">
<td align="center">rs9605903</td>
<td align="center">C</td>
<td align="center">T</td>
<td align="center">22</td>
</tr>
<tr class="odd">
<td align="center">rs5746647</td>
<td align="center">G</td>
<td align="center">T</td>
<td align="center">22</td>
</tr>
<tr class="even">
<td align="center">rs16980739</td>
<td align="center">T</td>
<td align="center">C</td>
<td align="center">22</td>
</tr>
</tbody>
</table>
<p>Number of rows in meta data = 15747</p>
<p>Number of rows/columns in genotype covariance = 15747, 15747</p>
</div>
<div id="idp-gwas" class="section level1">
<h1><span class="header-section-number">4</span> IDP GWAS</h1>
<p>We share the IDP GWAS results for all the IDPs being analyzed and the study cohort is the same as IDP model training (n = 24,490).</p>
<p>The GWAS results are by chromosome and by IDP and the naming rule is as follow:</p>
<ul>
<li><code>path-to-brainxcan_data/idp_gwas/[idp_type].[modality].chr[chr_num]/[idp_id]].parquet</code>
<ul>
<li><code>idp_type</code>: <code>residual</code> or <code>original</code></li>
<li><code>modality</code>: <code>dmri</code> or <code>t1</code></li>
<li><code>chr_num</code>: 1 to 22</li>
<li><code>idp_id</code>: IDP ID</li>
</ul></li>
</ul>
<p>The SNPs are in rows and columns are:</p>
<ul>
<li><code>variant_id</code>: rsID of the SNP</li>
<li><code>phenotype_id</code>: IDP ID</li>
<li><code>pval</code>: P-value of the association</li>
<li><code>b</code> and <code>b_se</code>: Estimated effect size and standard error</li>
<li><code>maf</code>: Minor allele frequency of the SNP in the GWAS cohort</li>
</ul>
<p>Further annotation of the SNPs can be found at <code>path-to-brainxcan_data/idp_gwas/snp_bim/chr[chr_num].bim</code>. This is essentially PLINK BIM file and the columns are: 1. Chromosome, 2. rsID, 3. placeholder, 4. Genomic position, 5. Effect allele, 6. Non-effect allele. <strong>The GWAS direction is the same as the one in BIM file.</strong></p>
<p>For instance, let’s load one GWAS file.</p>
<pre class="r"><code># set up the file path
idp_type = &#39;residual&#39;
modality = &#39;t1&#39;
chr_num = 22
idp_id = &#39;IDP-25002&#39; 
gwas_file = paste0(&#39;idp_gwas/&#39;, idp_type, &#39;.&#39;, modality, &#39;.chr&#39;, chr_num, &#39;/&#39;, idp_id, &#39;.parquet&#39;)
bim_file = paste0(&#39;idp_gwas/&#39;, &#39;snp_bim/&#39;, &#39;chr&#39;, chr_num, &#39;.bim&#39;)</code></pre>
<p>idp_type = <code>residual</code></p>
<p>modality = <code>t1</code></p>
<p>chr_num = <code>22</code></p>
<p>idp_id = <code>IDP-25002</code></p>
<p>Reading parquet file: <code>path-to-brainxcan_data/idp_gwas/residual.t1.chr22/IDP-25002.parquet</code></p>
<p>Reading BIM file: <code>path-to-brainxcan_data/idp_gwas/snp_bim/chr22.bim</code></p>
<pre class="r"><code>df_gwas = pd$read_parquet(paste0(datadir, &#39;/&#39;, gwas_file))
df_bim = read.table(
  paste0(datadir, &#39;/&#39;, bim_file), 
  col.names = c(&#39;chromosome&#39;, &#39;variant_id&#39;, &#39;ph&#39;, &#39;position&#39;, &#39;effect_allele&#39;, &#39;non_effect_allele&#39;), 
  stringsAsFactors = F
) %&gt;% select(-ph)  # remove placeholder column

# annotate GWAS results with additional SNP information
df_gwas = right_join(df_bim, df_gwas, by = &#39;variant_id&#39;)

df_gwas %&gt;% mutate(position = as.character(position)) %&gt;% head %&gt;% pander</code></pre>
<table>
<colgroup>
<col width="10%" />
<col width="10%" />
<col width="8%" />
<col width="12%" />
<col width="15%" />
<col width="11%" />
<col width="7%" />
<col width="8%" />
<col width="7%" />
<col width="7%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">chromosome</th>
<th align="center">variant_id</th>
<th align="center">position</th>
<th align="center">effect_allele</th>
<th align="center">non_effect_allele</th>
<th align="center">phenotype_id</th>
<th align="center">pval</th>
<th align="center">b</th>
<th align="center">b_se</th>
<th align="center">maf</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">22</td>
<td align="center">rs4911642</td>
<td align="center">16504399</td>
<td align="center">C</td>
<td align="center">T</td>
<td align="center">IDP-25002</td>
<td align="center">0.3064</td>
<td align="center">-0.02878</td>
<td align="center">0.02814</td>
<td align="center">0.08711</td>
</tr>
<tr class="even">
<td align="center">22</td>
<td align="center">rs7287144</td>
<td align="center">16886873</td>
<td align="center">G</td>
<td align="center">A</td>
<td align="center">IDP-25002</td>
<td align="center">0.7055</td>
<td align="center">0.003934</td>
<td align="center">0.01041</td>
<td align="center">0.2708</td>
</tr>
<tr class="odd">
<td align="center">22</td>
<td align="center">rs5748662</td>
<td align="center">16892858</td>
<td align="center">A</td>
<td align="center">G</td>
<td align="center">IDP-25002</td>
<td align="center">0.6634</td>
<td align="center">0.00454</td>
<td align="center">0.01043</td>
<td align="center">0.2699</td>
</tr>
<tr class="even">
<td align="center">22</td>
<td align="center">rs2027653</td>
<td align="center">16918335</td>
<td align="center">T</td>
<td align="center">C</td>
<td align="center">IDP-25002</td>
<td align="center">0.573</td>
<td align="center">-0.01169</td>
<td align="center">0.02074</td>
<td align="center">0.425</td>
</tr>
<tr class="odd">
<td align="center">22</td>
<td align="center">rs9680776</td>
<td align="center">17000277</td>
<td align="center">T</td>
<td align="center">C</td>
<td align="center">IDP-25002</td>
<td align="center">0.8555</td>
<td align="center">0.003334</td>
<td align="center">0.0183</td>
<td align="center">0.1795</td>
</tr>
<tr class="even">
<td align="center">22</td>
<td align="center">rs5747620</td>
<td align="center">17032698</td>
<td align="center">C</td>
<td align="center">T</td>
<td align="center">IDP-25002</td>
<td align="center">0.2179</td>
<td align="center">0.01388</td>
<td align="center">0.01127</td>
<td align="center">0.4</td>
</tr>
</tbody>
</table>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
