<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>An example BrainXcan run</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">BrainXcan software documentation</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/liangyy/brainxcan-docs">
    <span class="fab fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">An example BrainXcan run</h1>

</div>


<div id="about" class="section level1" number="1">
<h1><span class="header-section-number">1</span> About</h1>
<p>In this post, we run BrainXcan with an example.</p>
<p>First of all, please make sure you’ve cloned <a
href="https://github.com/hakyimlab/brainxcan">Brainxcan code
repository</a> and installed all the BrainXcan software
dependencies.</p>
<p>To perform BrainXcan analysis, we need to prepare the following
files:</p>
<ul>
<li>The BrainXcan database and meta files (download from <a
href="https://zenodo.org/record/4895174">brainxcan database zenodo
link</a>)
<ul>
<li>To run S-BrainXcan with permutation within LD blocks, you will also
need an LD block BED file, e.g. <a
href="https://bitbucket.org/nygcresearch/ldetect-data/src/master/EUR/fourier_ls-all.bed">link</a>
and we have it copied <a
href="../external_resources/data/fourier_ls-all.bed">here</a> as
well<br />
</li>
</ul></li>
<li>A file containing GWAS summary statistics. It should have the
following information:
<ul>
<li>SNP rsID (we use rsID as SNP identifier) and chromosome of the
SNP.</li>
<li>Effect and non-effect allele</li>
<li>Summary statistics (one in the two groups):
<ol style="list-style-type: decimal">
<li><strong>Group 1</strong>: effect size, standard error of effect size
(zero-centered)</li>
<li><strong>Group 2</strong>: z-score, GWAS sample size, allele
frequency</li>
</ol></li>
</ul></li>
<li>A pipeline configuration file.</li>
</ul>
</div>
<div id="downloading-brainxcan-database" class="section level1"
number="2">
<h1><span class="header-section-number">2</span> Downloading BrainXcan
database</h1>
<p>The BrainXcan database and meta files can be downloaded from <a
href="https://zenodo.org/record/4895174">brainxcan database zenodo
link</a>. The directory structure is</p>
<pre><code>brainxcan_data/
|-- bxcan_vis
|-- geno_covar
|-- idp_gwas
|-- idp_weights
`-- mr</code></pre>
<p>There is no need to change the structure and it is designed to be
used asis.</p>
<p>See some detailed notes on the BrainXcan database at “<a
href="database.html">Looking into BrainXcan database</a>”.</p>
</div>
<div id="downloading-gwas" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Downloading GWAS</h1>
<p>In this example, we provide a pre-formatted GWAS file at here: <a
href="https://uchicago.box.com/s/s2yg1gvxm5vv1q3yijf5h389o3hkadwv">preview
link</a> and <a
href="https://uchicago.box.com/shared/static/s2yg1gvxm5vv1q3yijf5h389o3hkadwv.gz">direct
download link</a>, which is adapted from a schizophrenia GWAS by <span
class="citation">Ripke et al. (2020)</span> (<a
href="https://www.medrxiv.org/content/10.1101/2020.09.12.20192922v1">medRxiv
link</a>).</p>
<p>First, let’s download and take a look at it.</p>
<pre><code># download
wget https://uchicago.box.com/shared/static/s2yg1gvxm5vv1q3yijf5h389o3hkadwv.gz \
  -O formatted.SCZ_PGC_2020.tsv.gz
  
# look at the first few rows
zcat &lt; formatted.SCZ_PGC_2020.tsv.gz | head

# returns
variant_id  chromosome  non_effect_allele   effect_allele   effect_size standard_error  zscore  sample_size frequency
rs3131972   chr1    G   A   -0.0020020026706730793  0.0111  -0.18439957 71533   0.1652
rs3131969   chr1    G   A   -0.006601743634562364   0.0118  -0.56070304 71533   0.1339
rs3131967   chr1    C   T   -0.005796768847177342   0.0119  -0.48651794 71533   0.1171
rs1048488   chr1    T   C   0.002696361547742533    0.0117  0.2303755   71533   0.1667
rs12562034  chr1    G   A   -0.003902375817241093   0.0156  -0.24843223 66899   0.1027
rs12124819  chr1    A   G   -0.011800104115750618   0.0181  -0.64967835 58941   0.2243
rs4040617   chr1    A   G   -0.0010993954433010642  0.0127  -0.09036144 71843   0.1295
rs4970383   chr1    C   A   0.02810116511090993 0.0111  2.5433347   64548   0.1964
rs4475691   chr1    C   T   0.012102946067745635    0.0121  0.9998151   65154   0.1429</code></pre>
</div>
<div id="preparing-the-brainxcan-configuration-file"
class="section level1" number="4">
<h1><span class="header-section-number">4</span> Preparing the BrainXcan
configuration file</h1>
<p><strong>TLDR</strong>: The configuration file is in YAML format and
the one for example run can be downloaded from <a
href="../external_resources/example_run/brainxcan_test.yaml">here</a>.
And there are a few fields that you need to fill in.</p>
<p>If you are not in a hurry, we recommend going through the following
sections which explain the configuration file in details.</p>
<p>The configuration file contains three main parts:</p>
<ul>
<li>Input/output file paths.</li>
<li>Meta information about the GWAS file.</li>
<li>Some parameter settings in BrainXcan.</li>
</ul>
<p>We will go over them one by one.</p>
<div id="specifying-inputoutput-paths" class="section level2"
number="4.1">
<h2><span class="header-section-number">4.1</span> Specifying
input/output paths</h2>
<ul>
<li><code>datadir</code>: Path to the BrainXcan database folder.</li>
<li><code>outdir</code>: Output directory name (if it does not exist, it
will be generated).</li>
<li><code>prefix</code>: Prefix of the output files (to distinguish
between results in the same output folder).</li>
<li><code>brainxcan_path</code>: Path to the BrainXcan code
repository.</li>
</ul>
<p>In the configuration YAML, we have:</p>
<pre><code>datadir: &#39;path-to-braixcan-database/brainxcan_data&#39;
outdir: &#39;brainxcan_test&#39;
prefix: &#39;SCZ&#39;
brainxcan_path: &#39;path-to-brainxcan-repository/brainxcan&#39;</code></pre>
</div>
<div id="specifying-gwas-meta-information" class="section level2"
number="4.2">
<h2><span class="header-section-number">4.2</span> Specifying GWAS meta
information</h2>
<p>Here we want to tell the software how to understand the GWAS
file.</p>
<ul>
<li><code>gwas</code>: Path to GWAS file (formats:
<code>csv(.gz)</code>, <code>tsv(.gz)</code>,
<code>parquet</code>).</li>
<li><code>snpid</code>: Column name of GWAS SNP rsID</li>
<li><code>effect_allele</code>: Column name of GWAS effect allele</li>
<li><code>non_effect_allele</code>: Column name of GWAS non-effect
allele</li>
<li><code>chr</code>: Column name of chromosome</li>
<li><code>position</code>: Column name of position</li>
</ul>
<p>Along with <strong>Group 1 or 2</strong> (see details at <a
href="#about">here</a>):</p>
<ul>
<li><strong>Group 1</strong>:
<ul>
<li><code>effect_size</code>: Column name of GWAS effect size</li>
<li><code>effect_size_se</code>: Column name of GWAS effect size
standard error</li>
</ul></li>
<li><strong>Group 2</strong>:
<ul>
<li>zscore: Column name of GWAS z-score</li>
<li>sample_size: Column name of GWAS sample size</li>
<li>allele_frequency: Column name of allele frequency</li>
</ul></li>
</ul>
<p>When both groups are specified, group 1 has high priority. For the
example run, we go with group 1.</p>
<p>In the configuration YAML, we have:</p>
<pre><code>gwas: &#39;formatted.SCZ_PGC_2020.tsv.gz&#39;
snpid: &#39;variant_id&#39;
effect_allele: &#39;effect_allele&#39;
non_effect_allele: &#39;non_effect_allele&#39;
chr: &#39;chromosome&#39;
effect_size: &#39;effect_size&#39;
effect_size_se: &#39;standard_error&#39;
position: &#39;position&#39;</code></pre>
</div>
<div id="specifying-parameters-in-brainxcan" class="section level2"
number="4.3">
<h2><span class="header-section-number">4.3</span> Specifying parameters
in BrainXcan</h2>
<p>There are several parameters underlying a BrainXcan run. They are
associated with three aspects:</p>
<ul>
<li>BrainXcan association test
<ul>
<li><code>model_type</code>: IDP prediction model type:
<code>ridge</code> or <code>elastic_net</code> (<strong>Default</strong>
= <code>ridge</code>)</li>
<li><code>idp_type</code>: IDP sets to use: <code>original</code> or
<code>residual</code> (after PC adjustment) (<strong>Default</strong> =
<code>residual</code>)</li>
<li><code>spearman_cutoff</code>: CV Spearman cutoff on models (only
models passing this criteria will be shown) (<strong>Default</strong> =
<code>0.1</code>)</li>
</ul></li>
<li>BrainXcan adjustment options
<ul>
<li>(<strong>Recommended</strong>) Adjusting z-scores with variance
control (The method details are described <a
href="https://www.biorxiv.org/content/10.1101/2023.10.17.562831v2">here</a>)
<ul>
<li>It requires GWAS effective sample size, GWAS trait heritability, and
pre-computed phi values for IDP models (IDP phi values can be downloaded
from <a href="../external_resources/data/idps-phi.txt">here</a>)</li>
<li><code>pheno_h2</code>: GWAS trait heritability (SNP heritability -
e.g. it can be estimated from LDSC regression)</li>
<li><code>gwas_n</code>: GWAS effective sample size</li>
<li><code>bxcan_vc_z</code>: Set to True when want run this
adjustment</li>
<li><code>bxcan_vc_phi</code>: Path to the phi value file</li>
</ul></li>
<li>(<strong>Not recommended</strong>) Adjusting z-scores with LD
block-based permutation
<ul>
<li><code>bxcan_ldblock_perm</code>: If want to run this adjustment, set
the path to the BED file (TAB-delimited base0 with header chr, start,
stop)</li>
<li><code>bxcan_ldblock_perm_seed</code>: Random seed</li>
<li><code>bxcan_ldblock_perm_nrepeat</code>: Number of permutation
across all IDPs</li>
<li><code>correction_factor_perm</code>: Correcting the standard error
of the simulated z-scores by multiplying with this factor (default =
1.1)</li>
</ul></li>
<li>(<strong>Not recommended</strong>) Adjusting z-scores via simulated
weights
<ul>
<li><code>bxcan_empirical_z</code>: Whether to run this adjustment or
not</li>
<li><code>bxcan_empirical_z_seed</code>: Random seed</li>
<li><code>bxcan_empirical_z_nrepeat</code>: Number of simulated
IDPs</li>
<li><code>correction_factor_emp</code>: Correcting the standard error of
the simulated z-scores by multiplying with this factor</li>
</ul></li>
</ul></li>
<li>Mendelian randomization (MR): extrating a list of significant IDP
and run MR
<ul>
<li><code>fdr_cutoff</code>: FDR cutoff to call an IDP significant
(<strong>Default</strong> = 0.1). This will overwrite
<code>signif_pval</code> below.</li>
<li><code>signif_pval</code>: P-value cutoff to call an IDP significant
(<strong>Default</strong> = <code>1e-5</code>)</li>
<li><code>signif_max_idps</code>: Maximun number of IDPs to run MR
(discard by p-value if too many IDPs passing p-value filter)
(<strong>Default</strong> = <code>10</code>)</li>
<li><code>ld_clump_yaml</code>: Parameters of LD clumping for defining
the instrument SNPs in MR (<strong>Default</strong> =
<code>{datadir}/mr/ld_clump.yaml</code>)<span
class="math inline">\(\dagger\)</span><br />
</li>
</ul></li>
<li>Dependent executables
<ul>
<li><code>rscript_exe</code>: Path to the Rscript executable
(<strong>Default</strong> = <code>Rscript</code>)</li>
<li><code>python_exe</code>: Path to the Python executable
(<strong>Default</strong> = <code>python</code>)</li>
<li><code>plink_exe</code>: Path to the plink v1.9 executable
(<strong>Default</strong> = <code>plink</code>)</li>
</ul></li>
</ul>
<p><span class="math inline">\(\dagger\)</span> <code>{datadir}</code>
will be interpreted as
<code>path-to-braixcan-database/brainxcan_data</code> internally since
we set
<code>datadir: path-to-braixcan-database/brainxcan_data</code>.</p>
<p>In the configuration YAML, we have:</p>
<pre><code>model_type: &#39;ridge&#39;
idp_type: &#39;residual&#39;
spearman_cutoff: 0.1 

fdr_cutoff: 0.1
signif_pval: 0.01
signif_max_idps: 5
gwas_pop: &#39;EUR&#39;
ld_clump_yaml: &#39;{datadir}/mr/ld_clump.yaml&#39;

rscript_exe: &#39;Rscript&#39; 
python_exe: &#39;python&#39;
plink_exe: &#39;plink&#39;</code></pre>
<p>The full list of parameters available in a BrainXcan configuration
YAML is explained (with default values) at <a
href="https://github.com/hakyimlab/brainxcan/blob/main/config.example.yaml">here</a></p>
</div>
</div>
<div id="running-brainxcan" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Running BrainXcan</h1>
<p>Now we are ready to run BrainXcan pipeline. The general command
is</p>
<pre><code># conda activate brainxcan 

REPO=path-to-brainxcan-repository
CONFIG=path-to-config-file

snakemake \
  -s $REPO/Snakefile \
  --configfile $CONFIG \
  [rule-name]</code></pre>
<p><code>[rule-name]</code> can take the following four options:</p>
<table>
<colgroup>
<col width="20%" />
<col width="14%" />
<col width="17%" />
<col width="27%" />
<col width="20%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">rule.name</th>
<th align="center">Association</th>
<th align="center">Visualization</th>
<th align="center">Mendelian.randomization</th>
<th align="center">Automated.report</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">SBrainXcanOnly</td>
<td align="center">x</td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
</tr>
<tr class="even">
<td align="center">SBrainXcanAndVis</td>
<td align="center">x</td>
<td align="center">x</td>
<td align="center"></td>
<td align="center"></td>
</tr>
<tr class="odd">
<td align="center">SBrainXcanAndMR</td>
<td align="center">x</td>
<td align="center"></td>
<td align="center">x</td>
<td align="center"></td>
</tr>
<tr class="even">
<td align="center">SBrainXcan</td>
<td align="center">x</td>
<td align="center">x</td>
<td align="center">x</td>
<td align="center">x</td>
</tr>
</tbody>
</table>
<!-- * `SBrainXcanOnly`: Run BrainXcan association only. -->
<!-- * `SBrainXcanAndVis`: Run BrainXcan association and generate visualization plots. -->
<!-- * `SBrainXcanAndMR`: Run BrainXcan association and Mendelian randomization on top hits. -->
<!-- * `SBrainXcan`: Run both BrainXcan and MR and generate visualization plots and an automated report.  -->
<p>Let’s say we want to run through everything, i.e. replacing
<code>[rule-name]</code> with <code>SBrainXcan</code>. So, we can
do:</p>
<pre><code>snakemake \
  -s $REPO/Snakefile \
  --configfile $CONFIG \
  -n -p \
  SBrainXcan</code></pre>
<p>We use <code>-n</code> options in <code>snakemake</code> to perform a
dry-run to see what will be run (<code>-p</code> is to print more
information about each step). The MR runs won’t be printed since we’ve
not known which IDPs will be significant but these will be updated
automatically during the computation.</p>
<p>If everything looks good, we could start running the command
(<code>-j1</code> tells snakemake to use one core).</p>
<pre><code>snakemake \
  -s $REPO/Snakefile \
  --configfile $CONFIG \
  -j1 -p \
  SBrainXcan</code></pre>
<p>It takes about 9 minutes with 4 Gb memory. The output files are put
in folder <code>brainxcan_test/</code> with prefix <code>SCZ</code>.</p>
</div>
<div id="looking-at-the-output-of-brainxcan" class="section level1"
number="6">
<h1><span class="header-section-number">6</span> Looking at the output
of BrainXcan</h1>
<p>The output files of the above run can be downloaded from <a
href="../external_resources/example_run/brainxcan_test.zip">here</a>.
Here is an overview of the files.</p>
<pre><code>brainxcan_test
├── SCZ.report.Rmd
├── SCZ.report.html
├── SCZ.sbrainxcan.csv
├── SCZ.vis.T1.pdf
├── SCZ.vis.dMRI.pdf
└── SCZ_files
    ├── MR_results
    │   ├── dmri_IDP-25687.MR.rds
    │   ├── dmri_IDP-25687.MR_sumstats.tsv
    │   ├── dmri_IDP-25687.MR_vis.pdf
    │   ├── dmri_PC-FA-TBSS-1.MR.rds
    │   ├── dmri_PC-FA-TBSS-1.MR_sumstats.tsv
    │   ├── dmri_PC-FA-TBSS-1.MR_vis.pdf
    │   ├── dmri_PC-ICVF-ProbTrack-1.MR.rds
    │   ├── dmri_PC-ICVF-ProbTrack-1.MR_sumstats.tsv
    │   ├── dmri_PC-ICVF-ProbTrack-1.MR_vis.pdf
    │   ├── dmri_PC-ICVF-TBSS-1.MR.rds
    │   ├── dmri_PC-ICVF-TBSS-1.MR_sumstats.tsv
    │   ├── dmri_PC-ICVF-TBSS-1.MR_vis.pdf
    │   ├── t1_IDP-25020.MR.rds
    │   ├── t1_IDP-25020.MR_sumstats.tsv
    │   └── t1_IDP-25020.MR_vis.pdf
    └── logs
        ├── dmri_IDP-25687.mr.log
        ├── dmri_IDP-25687.mr_vis.log
        └── [more MR log files]</code></pre>
<ul>
<li><code>[prefix].sbrainxcan.csv</code>: BrainXcan association results
in CSV format.</li>
<li><code>[prefix].vis.T1.pdf</code>: Visualization of associations of
structural IDPs.</li>
<li><code>[prefix].vis.dMRI.pdf</code>: Visualization of associations of
diffusion IDPs (only TBSS IDPs).</li>
<li><code>[prefix]_files/MR_results</code>: Mendelian randomization for
each significant IDPs
<ul>
<li><code>*.MR_vis.pdf</code>: Plotting effect sizes of instrument SNPs
in MR.</li>
<li><code>*.MR_sumstats.tsv</code>: Results of MR tests summarized in a
table (TSV format).</li>
<li><code>*.MR.rds</code>: Raw data used for MR test.</li>
</ul></li>
<li><code>[prefix].report.html</code>: The automated HTML report
summarizing the association and MR results (for diffusion IDPs, only
TBSS ones are shown).</li>
<li><code>[prefix]_files/logs/</code>: Log files of each analysis
module.</li>
</ul>
<div id="association-output-table" class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> Association output
table</h2>
<p>Let’s take a quick look at
<code>brainxcan_test/SCZ.sbrainxcan.csv</code> (first two rows). The
descriptions of each columns are at the bottom of the table.</p>
<p>Error in <code>dplyr::bind_rows()</code>: ! Can’t combine
<code>..1$bhat</code> <double> and <code>..2$bhat</code> <character>.
Backtrace: 1. … %&gt;% pander(split.cell = 2, emphasize.strong.rows = 3)
3. dplyr::bind_rows(…)</p>
</div>
<div id="mendelian-randomization-output-table" class="section level2"
number="6.2">
<h2><span class="header-section-number">6.2</span> Mendelian
randomization output table</h2>
<p>Let’s take a quick look at
<code>brainxcan_test/SCZ_files/MR_results/dmri_PC-FA-TBSS-1.MR_sumstats.tsv</code>
(first two rows). The descriptions of each columns are at the bottom of
the table.</p>
<table>
<colgroup>
<col width="12%" />
<col width="9%" />
<col width="9%" />
<col width="11%" />
<col width="10%" />
<col width="15%" />
<col width="15%" />
<col width="15%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">direction</th>
<th align="center">method</th>
<th align="center">nsnp</th>
<th align="center">b</th>
<th align="center">pval</th>
<th align="center">Q</th>
<th align="center">Q_df</th>
<th align="center">Q_pval</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">IDP -&gt; Phenotype</td>
<td align="center">MR Egger</td>
<td align="center">48</td>
<td align="center">0.105</td>
<td align="center">0.462</td>
<td align="center">228</td>
<td align="center">46</td>
<td align="center">6.66e-26</td>
</tr>
<tr class="even">
<td align="center">IDP -&gt; Phenotype</td>
<td align="center">Weighted median</td>
<td align="center">48</td>
<td align="center">0.0324</td>
<td align="center">0.258</td>
<td align="center">NA</td>
<td align="center">NA</td>
<td align="center">NA</td>
</tr>
<tr class="odd">
<td align="center"><strong>MR direction</strong></td>
<td align="center"><strong>MR method</strong></td>
<td align="center"><strong>Number of SNPs used in MR</strong></td>
<td align="center"><strong>Slope in MR</strong></td>
<td align="center"><strong>P-value in MR</strong></td>
<td align="center"><strong>Heterogenity test Q</strong></td>
<td align="center"><strong>Heterogenity test degree of
freedom</strong></td>
<td align="center"><strong>Heterogenity test p-value</strong></td>
</tr>
</tbody>
</table>
</div>
<div id="visualization-and-report" class="section level2" number="6.3">
<h2><span class="header-section-number">6.3</span> Visualization and
report</h2>
<ul>
<li><a
href="../external_resources/example_run/brainxcan_test/SCZ_PGC_2020.vis.T1.pdf">Visualization
of structural IDPs</a></li>
<li><a
href="../external_resources/example_run/brainxcan_test/SCZ_PGC_2020.vis.dMRI.pdf">Visualization
of diffusion IDPs</a></li>
<li><a
href="../external_resources/example_run/brainxcan_test/SCZ_PGC_2020.report.html">Automated
report</a></li>
</ul>
<p>We provide interactive HTMLs showing regions in the
visualization:</p>
<ul>
<li><a href="../external_resources/pages/label_Cortical.html">Cortical
gray matters</a></li>
<li><a
href="../external_resources/pages/label_Subcortical-GMvol.html">Subcortical
gray matters</a></li>
<li><a
href="../external_resources/pages/label_Subcortical-vol.html">Subcortical
total volumes</a></li>
<li><a
href="../external_resources/pages/label_Cerebellum.html">Cerebellum gray
matters</a></li>
<li><a href="../external_resources/pages/label_TBSS.html">TBSS-basd
dMRI</a></li>
</ul>
<p>These HTMLs can be generated along the BrainXcan visualization by
setting <code>bxcan_region_vis=True</code> in the configuration
(default: False). The output files will be put inside the folder called
<code>[prefix].vis.region_vis/</code>.</p>
</div>
</div>
<div id="extra-notes-on-using-snakemake" class="section level1"
number="7">
<h1><span class="header-section-number">7</span> Extra notes on using
<code>snakemake</code></h1>
<p>The BrainXcan software is essentially a data pipeline containing
multiple steps and intermediate files. It is implemented using
<code>snakemake</code> <span class="citation">Mölder et al.
(2021)</span> and the features provided by <code>snakemake</code> are
directly applicable here.</p>
<p>For instance, the parameters in the configuration file can be
replaced by <code>--config</code>, e.g. if we want to change the prefix
of the output, we can do it without changing the config file:</p>
<pre><code>snakemake \
  -s $REPO/Snakefile \
  --configfile $CONFIG \
  -j1 -p \
  SBrainXcan \
  --config prefix=another_test</code></pre>
<p>More over, if the intermediate files are already present, snakemake
will not re-generate them. In other word, if you run
<code>SBrainXcanOnly</code> first and then run
<code>SBrainXcanAndVis</code>, it won’t run BrainXcan association again.
If you want to enforce re-runing, use the flag
<code>--forceall</code>.</p>
<p>BrainXcan software is written with snakemake v6. For more cool
features of snakemake, please refer to <a
href="https://snakemake.readthedocs.io/en/stable/">snakemake
documentation</a>.</p>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="unnumbered">References</h1>
<div id="refs" class="references csl-bib-body hanging-indent"
entry-spacing="0">
<div id="ref-molder2021sustainable" class="csl-entry">
Mölder, Felix, Kim Philipp Jablonski, Brice Letcher, Michael B Hall,
Christopher H Tomkins-Tinch, Vanessa Sochat, Jan Forster, et al. 2021.
<span>“Sustainable Data Analysis with Snakemake.”</span>
<em>F1000Research</em> 10.
</div>
<div id="ref-ripke2020mapping" class="csl-entry">
Ripke, Stephan, James TR Walters, Michael C O’Donovan, Schizophrenia
Working Group of the Psychiatric Genomics Consortium, et al. 2020.
<span>“Mapping Genomic Loci Prioritises Genes and Implicates Synaptic
Biology in Schizophrenia.”</span> <em>MedRxiv</em>.
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
